satellite/metainfo/metainfo.go

```diff
@@ -502,35 +528,69 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 		if allSizesValid {
 			redundancy, err := eestream.NewRedundancyStrategyFromProto(pointer.GetRemote().GetRedundancy())
 			if err != nil {
-				return Error.Wrap(err)
+				endpoint.log.Debug("pointer contains an invalid redundancy strategy", zap.Error(Error.Wrap(err)))
+				return rpcstatus.Error(rpcstatus.InvalidArgument, "invalid redundancy strategy; MinReq and/or Total are invalid")
```

Comment:
Why don't we add the `err` to the error message here?

Reply:
I don't like the idea of sending to client errors which I don't know which information contains, they could leak some sensitive information.

Reply:
Not sure what sensitive info can leak here. But the original message could really help the client figure out easier what was really wrong with the redundancy config.

Reply:
It isn't a good practice to return error stack traces to clients, such information could lead to finding bugs that could have a security impact. On the other hand, the stack traces of the server for a client error (this is an invalid parameter sent by the client) shouldn't be needed to report anything; internal errors is another story, however I usually prefer not leaking stack traces or other information like panics, etc.

Storj is an open-source project and because of that, the stack traces are OK to be returned because anyway everybody has access to the code.

I chose a conservative approach for it, but I'm not against to change it after I gave the reason for it.

Apart from that, for this specific case.

* Some error message is coming from infections dependency and it doesn't provide a good message for the user https://github.com/vivint/infectious/blob/HEAD/fec.go#L53 however, there are others which could be useful (the ones returned by https://github.com/storj/storj/blob/aba9a9028f9f70ce976822827abf492b121d4357/uplink/eestream/encode.go#L67:6)
* I'm seeing that we could add the error message to rpc error message to complement the one given, however, I didn't find a way with `errs` package to only print the message without the stack trace. It's possible to have a minor change in `errs` to only return the stack traces when `%v` (or another one alike) is used in `printf` functions family, but that's something that we could discuss in another moment.

Let me know if I add the error to the message, happy to do it.

Reply:
It's about returning the error message only, not print the stack traces.

Wouldn't `err.Error()` return the error message of `err`?

Reply:
What I understood in the `errs` implementation it would print the stack trace, but I'm going to have a try to ensure it.

Reply:
Stack traces are printed only in this case: `fmt.Printf("%+v\n", err)`

Reply:
OK, cool, then it's fine.

Reply:
I took a look at the implementation. I thought that it didn't was in that way because it isn't actually the `%+v` which prints the strack traces, it's just only the `+`, if you do `fmt.Pritnf("%+s", err)` then the stack traces are printed too.